src/engine/Server.h

```diff
@@ -118,21 +120,53 @@ class Server {
   Awaitable<void> process(
       const ad_utility::httpUtils::HttpRequest auto& request, auto&& send);
 
-  /// Handle a http request that asks for the processing of a query.
+  /// Handle a http request that asks for the processing of an operation.
+  /// This is only a wrapper for `processQuery` and `processUpdate` which
+  /// does the error handling.
   /// \param params The key-value-pairs  sent in the HTTP GET request.
-  /// \param query The query.
+  /// \param operation The operation.
   /// \param requestTimer Timer that measure the total processing
   ///                     time of this request.
   /// \param request The HTTP request.
   /// \param send The action that sends a http:response (see the
   ///             `HttpServer.h` for documentation).
   /// \param timeLimit Duration in seconds after which the query will be
   ///                  cancelled.
+  template <bool isQuery>
```

Comment:
Not ideal. Maybe do this as a function reference instead?

Reply:
Do you have a nice solution for this? I wasn't able to just pass the processing function in as a parameter with `std::function`. It worked with a lambda, but was very verbose.

Reply:
std::function does only work if you very very verbosely given the types of `request` and `send` etc. names.
And you still need to basically pass in a lambda and it is messy.
I think for now you can just leave the bool because if (with my above suggestion) 
the function is called `processQueryOrUpdate`  and has a parameter `qeryOrUpdate` then the bool `isQuery` is perfectly understandable to humans.

Reply:
Okay then we'll stick with the general design. I have replaced the bool with an enum. This makes it completely self-explanatory.